compiler.py

```diff
@@ -314,12 +314,12 @@ def expose_exp(self, e: expr) -> expr:
                     If(
                         Compare(
                             BinOp(
-                                GlobalValue(Label("free_ptr")),
+                                GlobalValue(Label("free_ptr").name),
```

Comment:
It would be better if `GlobalValue` took a `Label` argument instead of a `str`.
Printing of `GlobalValue` has to be changed to use `.name`.

Reply:
Done